### DISCUSSION TITLE
Ffmpeg/AVPacket question

### DISCUSSION THREAD

**Discussion by BigDog6432** [bytedeco/javacpp-presets#1024] [Q&A] [answered]
I am trying to create my own AVPackets out of a Java ByteBuffer but I seem to be getting an “Exception_Access_Violation” error.  I am using the ReadFewFrames as a starting example.

The video that I am using contains both video and data packets.  I would like to replace the data packets with my own packets.

ByteBuffer bb = new ByteBuffer();
byte [] data =  new byte[] {0x00, 0x01, ...};
bb.put(data);

BytePointer bp = new BytePointer(bb);
AVPacket av = new AVPacket(bp);

I am copying the following properties from the original data packets into my av packet: dts, duration, flags, pts, size and stream_index.

I then am calling a format.av_interleaved_write_frame passing in my context and av.

and then freeing the packet which I created.
av_packet_unref(av),

When I run my program it dies almost immediately (it looks to be having an issue inside of acutely.dll).  When I comment out my code and write and free the original packet the program works fine.  Does anybody have a example of doing a something similar that they could share I would greatly appreciate it, or any idea if I am missing a magic line of could.

Thanks in advance

J. Stoy

**Comment by saudet** [comment, marked as answer]
I think you may want to put your data in the packet by calling AVPacket.data(bp)? Calling AVPacket(bp) isn't going to do that.

**Comment by BigDog6432** [comment]
Samuel,

I made that change. The program completes successfully.  Thanks Samuel!
